src/DocumentDB.ChangeFeedProcessor/Utils/TaskExtensions.cs

```diff
@@ -28,5 +28,23 @@ public static async Task IgnoreException(this Task task)
                 // ignore
             }
         }
+
+        public static async Task<TResult> AbortOnTimeout<TResult>(this Task<TResult> responseTask, TimeSpan timeout, CancellationToken token)
+        {
+            using (CancellationTokenSource cancelSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            {
+                Task timeoutTask = Task.Delay(timeout, cancelSource.Token);
+
+                if (responseTask == await Task.WhenAny(responseTask, timeoutTask).ConfigureAwait(false))
+                {
+                    cancelSource.Cancel(); // cancel the delay task
+                    return responseTask.GetAwaiter().GetResult(); // propagate exception
+                }
+            }
+
+            // timeout or cancellation token has fired
+            token.ThrowIfCancellationRequested();
+            throw new TimeoutException();
```

Comment:
Can we put some text to the TimeoutException so that we can recognize later it's coming from CFP?

Reply:
sure, will do